src/fork.rs

```diff
@@ -187,11 +180,29 @@ impl ForkStorage {
     }
 }
 
+/// Trait that provides necessary data when
+/// forking a remote chain.
+/// The method signatures are similar to methods from ETHNamespace and ZKNamespace.
+pub trait ForkSource {
+    /// Returns the Storage value at a given index for given address.
```

Comment:
We should add more thorough comments for the parameters and descriptions for all 4 methods in this `pub trait` (as they will show up in the docs)